### PR TITLE
[SPARK-44153][CORE][UI][FOLLOWUP] Use JAVA_HOME and stderr

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This is a follow-up of https://github.com/apache/spark/pull/41709 to address the review comments.

### Why are the changes needed?

1. Use `JAVA_HOME` prefixed `jmap` to ensure the same version's `JVM` and JMAP.
2. Use the existing stderr instead of merging `stderr` and `stdout` via `redirectErrorStream`
3. Use `tryWithResource`.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual review.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
cc @pan3793 and @mridulm

File: core/src/main/scala/org/apache/spark/util/Utils.scala [hunk 1]
```diff
@@ -2291,15 +2291,16 @@ private[spark] object Utils extends Logging with SparkClassUtils {
...
+    val jmap = System.getProperty("java.home") + "/bin/jmap"
+    val builder = new ProcessBuilder(jmap, "-histo:live", pid)
     val p = builder.start()
-    val r = new BufferedReader(new InputStreamReader(p.getInputStream()))
     val rows = ArrayBuffer.empty[String]
-    var line = ""
-    while (line != null) {
-      if (line.nonEmpty) rows += line
-      line = r.readLine()
+    Utils.tryWithResource(new BufferedReader(new InputStreamReader(p.getInputStream()))) { r =>
```
Comment (dongjoon-hyun) [inline core/src/main/scala/org/apache/spark/util/Utils.scala:2298]: @mridulm . I'll make another PR to limit the number of results and to address your comment for readLine parts. In this PR, I just wrapped with `tryWithResource`.

**Review by yaooqinn — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]
@dongjoon-hyun thanks for fixing this issue.

**Comment by dongjoon-hyun** [comment]
Thank you for reviewing and fixing the issue, @pan3793 and @yaooqinn .
I didn't realized the level of severity at first glance. Sorry for that. Thank you for your help.

**Comment by dongjoon-hyun** [comment]
Merged to master!

**Pull request closed without merging**